arm-datalake-analytics/job/2016-11-01/swagger/job.json

```diff
@@ -0,0 +1,1070 @@
+      "allOf": [
+        {
+          "$ref": "#/definitions/JobProperties"
+        }
+      ],
+      "properties": {
+        "resources": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobResource"
+          },
+          "description": "the list of resources that are required by the job"
+        },
+        "statistics": {
+          "$ref": "#/definitions/JobStatistics",
+          "description": "the job specific statistics."
+        },
+        "debugData": {
+          "$ref": "#/definitions/JobDataPath",
+          "description": "the job specific debug data locations."
+        },
+        "diagnostics": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Diagnostics"
+          },
+          "description": "the diagnostics for the job."
+        },
+        "algebraFilePath": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the algebra file path after the job has completed"
+        },
+        "totalCompilationTime": {
+          "type": "string",
+          "format": "duration",
+          "readOnly": true,
+          "description": "the total time this job spent compiling. This value should not be set by the user and will be ignored if it is."
+        },
+        "totalPauseTime": {
+          "type": "string",
+          "format": "duration",
+          "readOnly": true,
+          "description": "the total time this job spent paused. This value should not be set by the user and will be ignored if it is."
+        },
+        "totalQueuedTime": {
+          "type": "string",
+          "format": "duration",
+          "readOnly": true,
+          "description": "the total time this job spent queued. This value should not be set by the user and will be ignored if it is."
+        },
+        "totalRunningTime": {
+          "type": "string",
+          "format": "duration",
+          "readOnly": true,
+          "description": "the total time this job spent executing. This value should not be set by the user and will be ignored if it is."
+        },
+        "rootProcessNodeId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the ID used to identify the job manager coordinating job execution. This value should not be set by the user and will be ignored if it is."
+        },
+        "yarnApplicationId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "the ID used to identify the yarn application executing the job. This value should not be set by the user and will be ignored if it is."
+        },
+        "yarnApplicationTimeStamp": {
+          "type": "integer",
+          "readOnly": true,
+          "format": "int64",
+          "description": "the timestamp (in ticks) for the yarn application executing the job. This value should not be set by the user and will be ignored if it is."
+        },
+        "compileMode": {
```

Comment:
It is not required. If it is not specified the server will chose the best mode depending on the current server load and the job type. I will add this information.